### PR TITLE
fix: FormHandler handle form deletion properly

### PR DESCRIPTION
**Pull request by panagiotisgts**
## Description

The entity ID to handle deletion of Forms is derived from the formKey present in the record and not the record key. 

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (ChrisKujawa) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:82]: ❓ Not clear to me why isDeleted was not published before?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (ChrisKujawa) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:89]: ❌ Doesn't it mean we don't update the other properties then?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (ChrisKujawa) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:88]: ❓ Why we have here a different ID then?

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]
> The entity ID to handle deletion of Forms is derived from the formKey present in the record and not the record key.

Sorry @panagiotisgts I feel I miss some context. The changes are not fully clear to me. Can you comment on my questions please.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (panagiotisgts) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:89]: For Forms specifically nothing gets updated apart from the `isDeleted` tag. Every form ingested has it's own key but carries the same `formId` and get's a version increment.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (panagiotisgts) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:88]: The change in ID is for the upsert to point to the previous Form record. Form records are not being updated, but overriden by providing a version increment.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (panagiotisgts) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:82]: It was published before, but it created a new record marked as `deleted` with a version increment.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java
Comment (ChrisKujawa) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FormHandler.java:82]: ❓ I guess we don't need these changes then right ?

**Review by ChrisKujawa — APPROVED** [review]

**Comment by ChrisKujawa** [comment]
Fyi i disabled the identity test to unblock us here as it was blocking the merge. In general that seemed to be an issue on main in general and on other PRs 

See https://camunda.slack.com/archives/C071KP5BTHB/p1734643817690859?thread_ts=1734606811.491299&cid=C071KP5BTHB

**Pull request merged**
